solr/core/src/java/org/apache/solr/highlight/DefaultSolrHighlighter.java

```diff
@@ -695,7 +695,7 @@ protected Object doHighlightingByHighlighter(
 
     // Try term vectors, which is faster
     //  note: offsets are minimally sufficient for this HL.
-    final Fields tvFields = schemaField.storeTermOffsets() ? reader.getTermVectors(docId) : null;
+    final Fields tvFields = schemaField.storeTermOffsets() ? reader.termVectors().get(docId) : null;
```

Comment:
we can pass `reader.termVectors()`
we are not using any other method of `reader`

Reply:
I note that this is a `protected` method, but I don't see any classes that override this? Any potential issues for plugins if we changed the method signature? (either easy issues like plugins would need to update the method signature, or thornier issues like plugins actually use methods on the reader?)

Reply:
we are only using the `doHighlightingByHighlighter` method here: https://github.com/apache/solr/blob/67be5d0b01bd653f7315b93af728d2259bc9daf2/solr/core/src/java/org/apache/solr/highlight/DefaultSolrHighlighter.java#L572
So, won't it work if we pass `reader.termVectors()` instead of `reader`?